rllib/examples/rl_modules/classes/custom_action_distribution_rlm.py

```diff
@@ -0,0 +1,153 @@
+    `TorchCategoricalWithTemp` class for details.
+
+    .. testcode::
+
+        import numpy as np
+        import gymnasium as gym
+
+        my_net = CustomActionDistributionRLModule(
+            observation_space=gym.spaces.Box(-1.0, 1.0, (4,), np.float32),
+            action_space=gym.spaces.Discrete(4),
+            model_config={"action_dist_temperature": 5.0},
+        )
+
+        B = 10
+        data = torch.from_numpy(
+            np.random.random_sample(size=(B, 4)).astype(np.float32)
+        )
+        # Expect a relatively high-temperature distribution.
+        # Set "action_dist_temperature" to small values << 1.0 to approximate greedy
+        # behavior (even when stochastically sampling from the distribution).
+        print(my_net.forward_exploration({"obs": data}))
+    """
+
+    @override(TorchRLModule)
+    def setup(self):
+        """Use this method to create all the model components that you require.
+
+        Feel free to access the following useful properties in this class:
+        - `self.model_config`: The config dict for this RLModule class,
+        which should contain flexible settings, for example: {"hiddens": [256, 256]}.
+        - `self.observation|action_space`: The observation and action space that
+        this RLModule is subject to. Note that the observation space might not be the
+        exact space from your env, but that it might have already gone through
+        preprocessing through a connector pipeline (for example, flattening,
+        frame-stacking, mean/std-filtering, etc..).
+        - `self.inference_only`: If True, this model should be built only for inference
+        purposes, in which case you may want to exclude any components that are not used
+        for computing actions, for example a value function branch.
+        """
+        input_dim = self.observation_space.shape[0]
+        hidden_dim = self.model_config.get("hidden_dim", 256)
+        output_dim = self.action_space.n
+
+        # Define simple encoder, and policy- and vf heads.
+        self._encoder = torch.nn.Sequential(
+            torch.nn.Linear(input_dim, hidden_dim),
+            torch.nn.ReLU(),
+        )
+        self._policy_net = torch.nn.Linear(hidden_dim, output_dim)
+        self._vf = nn.Linear(hidden_dim, 1)
+
+        # Plug in a custom action dist class.
+        # NOTE: If you need more granularity as to which distribution class is used by
+        # which forward method (forward_inference, forward_explration, forward_train),
+        # override the RLModule methods `get_inference_action_dist_cls`,
+        # `get_exploration_action_dist_cls`, and `get_train_action_dist_cls` and return
+        # your custom class(es) from these. In this case, leave self.action_dist_cls set
+        # to None, its default value.
+        self.action_dist_cls = _make_categorical_with_temperature(
+            self.model_config.get("action_dist_temperature", 1.0),
+        )
```

Comment:
@sven1977 I think, for the purpose of this PR, using this API `self.action_dist_cls` still makes sense.
But I'd also argue that we should make a (backward-compatible) change to RL Modules:
`RLModule.get_inference_action_dist_cls` should be a getter method `RLModule.inference_action_dist_cls` to make it look like an attribute but does the same thing as today. If user now wants to override, they set that attribute in the setup method. Because today, we have a mixture of attributes and these getter methods to modify RLModules.

That way, the default way to change all action distributions would be the setup method, while the old way of overriding `RLModule.get_inference_action_dist_cls` would still be available by overriding the `RLModule.inference_action_dist_cls` getter method.